pkg/storegateway/bucket_index_reader.go

```diff
@@ -311,10 +311,12 @@ func (r *bucketIndexReader) fetchPostings(ctx context.Context, keys []labels.Lab
 	for ix, key := range keys {
 		// Get postings for the given key from cache first.
 		if b, ok := fromCache[key]; ok {
-			stats.postingsTouched++
-			stats.postingsTouchedSizeSum += len(b)
+			stats.update(func(stats *queryStats) {
```

Comment:
Is there any concern that we are adding locking here where there previously wasn't any? Should this be fast since it's uncontended?

Reply:
I'm not much concerned because the number of entries in `keys` is very small. The only concern I had (which was when we iterate through postings) has been addressed without locking.